src/dbos-executor.ts

```diff
@@ -347,58 +351,144 @@ export class DBOSExecutor {
+    this.workflowInfoMap.set(wfn, workflowInfo);
+    this.logger.debug(`Registered workflow ${wfn}`);
   }
 
-  #registerTransaction<T extends any[], R>(txn: Transaction<T, R>, params: TransactionConfig = {}) {
-    if (this.transactionInfoMap.has(txn.name)) {
-      throw new DBOSError(`Repeated Transaction name: ${txn.name}`);
+  #registerTransaction(ro: MethodRegistrationBase) {
+    const txf = ro.registeredFunction as Transaction<unknown[], unknown>;
+    const tfn = getRegisteredMethodClassName(txf) + '.' + ro.name;
+
+    if (this.transactionInfoMap.has(tfn)) {
+      throw new DBOSError(`Repeated Transaction name: ${tfn}`);
     }
     const txnInfo: TransactionInfo = {
-      transaction: txn,
-      config: params,
+      transaction: txf,
+      config: {...ro.txnConfig},
     };
-    this.transactionInfoMap.set(txn.name, txnInfo);
+    this.transactionInfoMap.set(tfn, txnInfo);
+    this.logger.debug(`Registered transaction ${tfn}`);
   }
 
-  #registerCommunicator<T extends any[], R>(comm: Communicator<T, R>, params: CommunicatorConfig = {}) {
-    if (this.communicatorInfoMap.has(comm.name)) {
-      throw new DBOSError(`Repeated Commmunicator name: ${comm.name}`);
+  #registerCommunicator(ro: MethodRegistrationBase) {
+    const comm = ro.registeredFunction as Communicator<unknown[], unknown>;
+    const cfn = getRegisteredMethodClassName(comm) + '.' + ro.name;
+    if (this.communicatorInfoMap.has(cfn)) {
+      throw new DBOSError(`Repeated Commmunicator name: ${cfn}`);
     }
     const commInfo: CommunicatorInfo = {
       communicator: comm,
-      config: params,
+      config: {...ro.commConfig},
     };
-    this.communicatorInfoMap.set(comm.name, commInfo);
+    this.communicatorInfoMap.set(cfn, commInfo);
+    this.logger.debug(`Registered communicator ${cfn}`);
   }
 
-  async workflow<T extends any[], R>(wf: Workflow<T, R>, params: InternalWorkflowParams, ...args: T): Promise<WorkflowHandle<R>> {
+  getWorkflowInfo(wf: Workflow<unknown[], unknown>) {
+    const wfname = (wf.name === DBOSExecutor.tempWorkflowName)
+      ? wf.name
+      : getRegisteredMethodClassName(wf) + '.' + wf.name;
+    return this.workflowInfoMap.get(wfname);
+  }
+  getWorkflowInfoByStatus(wf: WorkflowStatus) {
+    const wfname = wf.workflowClassName + '.' + wf.workflowName;
+    let wfInfo = this.workflowInfoMap.get(wfname);
+    if (!wfInfo && !wf.workflowClassName) {
+      for (const [_wfn, wfr] of this.workflowInfoMap) {
+        if (wf.workflowName === wfr.workflow.name) {
+          if (wfInfo) {
+            throw new DBOSError(`Recovered workflow function name '${wf.workflowName}' is ambiguous.  The ambiguous name was recently added; remove it and recover pending workflows before re-adding the new function.`);
+          }
+          else {
+            wfInfo = wfr;
+          }  
+        }
+      }
+    }
+
+    return {wfInfo, config: getClassConfiguration(wf.workflowClassName, wf.workflowConfigName)};
```

Comment:
`config` is a bit overloaded now and we need to be more precise -- we also have WorkflowConfig, TransactioinConfig, and CommunicatorConfig..

```suggestion
    return {wfInfo, configuredClass: getConfiguredClass(wf.workflowClassName, wf.workflowConfigName)};
```

Reply:
I can change the names a bit, but not just one line in isolation as suggested.